hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOzoneSnapshotRestore.java

```diff
@@ -258,9 +259,24 @@ public void testRestoreSnapshotDifferentBucket(BucketLayout bucketLayoutTest)
     int volBucketKeyCount = keyCount(buck, snapshotKeyPrefix + keyPrefix);
     Assertions.assertEquals(5, volBucketKeyCount);
 
+    // Delete keys from the source bucket.
```

Comment:
Why do you need to delete the keys from the source bucket ?

Reply:
Problem is that RocksDB's seek operation doesn't 100% guarantee that item key is available. Tho we believe that it is. When we check if there is a directory for the key name, sometime it returns non-null response but in reality item doesn't exist.
[Here](https://github.com/apache/ozone/blob/d8765436c2bf76547973cc568f1648f1b1fe9fcb/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java#L1184) sometimes seek returns null and sometimes it return source bucket (which is wrong). When is returns source bucket, we create fake dir and return the response.

I was unsure what should be the correct behavior for this so I went ahead to delete keys from source bucket for now as temporary fix. But we should revisit the `createFakeDirIfShould` logic and fix it there. This is the original PR: https://github.com/apache/ozone/pull/3774/files

Reply:
fyi `createFakeDirIfShould` was brought in 2 months ago as a test flakiness fix: #4038

Reply:
I agree with @smengcl  on checkout intention of `createFakeDirIfShould` and that's why I went ahead by deleting the keys from source bucket because it doesn't matter for that test if keys are present in source bucket. The requirement for test is that keys should be present in snapshot path (snapshotKeyPrefix).

Reply:
Thanks @hemantk-12 . Just for the record, HDDS-7870/HDDS-7871 would be the follow-up to properly fix the issue in the master branch.